components/CreateTransaction.js

```diff
@@ -0,0 +1,109 @@
+import styled from "styled-components";
+
+export default function Form({ onSubmit, submitting = false }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const form = event.currentTarget;
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    data.amount = Number(data.amount);
+
+    onSubmit(data);
+
+    form.reset();
+    form.querySelector("#name")?.focus();
+  }
+  const today = new Date().toISOString().slice(0, 10);
+  return (
+    <>
+      <HeaderForm>Please fill out all fields</HeaderForm>
+      <FormContainer onSubmit={handleSubmit}>
+        <Label htmlFor="name">Name</Label>
+        <Input
+          id="name"
+          name="name"
+          type="text"
+          placeholder="Please add your name"
+          required
+        />
+        <Label htmlFor="amount">Amount</Label>
+        <Input
+          id="amount"
+          name="amount"
+          type="number"
+          placeholder="Please add amount"
+          required
+        />
+        <select id="category" name="category" required defaultValue="">
+          <option value="" disabled>
+            --choose category--
+          </option>
+          <option>Education</option>
+          <option>Entertainment</option>
+          <option>Groceries</option>
+          <option>Health</option>
+          <option>Insurance</option>
+          <option>Investment</option>
+          <option>Miscellaneous</option>
+          <option>Rent</option>
+          <option>Restaurants</option>
+          <option>Salary</option>
+          <option>Savings</option>
+          <option>Transportation</option>
+          <option>Utilities</option>
```

Comment:
maybe this is now a good chance to make use of your categories collection API endpoint and map over the received data.